Raven.Voron/Voron/Platform/Win32/Win32MemoryMapPager.cs

```diff
@@ -363,15 +361,36 @@ public override void MaybePrefetchMemory(List<Page> sortedPages)
 
             fixed (Win32MemoryMapNativeMethods.WIN32_MEMORY_RANGE_ENTRY* entries = list.ToArray())
             {
-                Win32MemoryMapNativeMethods.PrefetchVirtualMemory(Win32NativeMethods.GetCurrentProcess(),
+                Win32MemoryMapNativeMethods.PrefetchVirtualMemory(_currentProcess,
                     (UIntPtr)list.Count,
                     entries, 0);
             }
         }
 
+        public override void MaybePrefetchMemory(List<long> pagesToPrefetch)
+        {
+            if (CanPrefetch == false)
+                return; // not supported
+
+            if (pagesToPrefetch.Count == 0)
+                return;
+
+            var entries = new Win32MemoryMapNativeMethods.WIN32_MEMORY_RANGE_ENTRY[pagesToPrefetch.Count];
+            for (int i = 0; i < entries.Length; i++)
+            {
+                entries[i].NumberOfBytes = (IntPtr)(4 * PageSize);
```

Comment:
We can comment that, sure. We dont have much information besides what is provided by Mark Russinovsky in Windows Internals 6th edition. He clearly states it is a hint to the kernel, not a requirement. That is why if the kernel believes it shouldnt do it, it wont. And also if it believes it has to get more data that what you asked for, he will do that too.